src/sentry/tasks/unmerge.py

```diff
@@ -430,32 +450,48 @@ def collect_tsdb_data(caches, project, events):
                 caches["Release"](project.organization_id, release).id,
             )
 
-            frequencies[event.datetime][TSDBModel.frequent_releases_by_group][event.group_id][
-                grouprelease.id
+            frequencies[event.datetime][TSDBModel.frequent_releases_by_group][str(event.group_id)][
+                str(grouprelease.id)
             ] += 1
 
     return counters, sets, frequencies
 
 
-def repair_tsdb_data(caches, project, events):
+def repair_tsdb_data(
+    caches: Mapping[str, Any], project: Project, events: Sequence[GroupEvent]
+) -> None:
     counters, sets, frequencies = collect_tsdb_data(caches, project, events)
 
     for timestamp, data in counters.items():
         for model, keys in data.items():
             for (key, environment_id), value in keys.items():
                 tsdb.backend.incr(model, key, timestamp, value, environment_id=environment_id)
 
-    for timestamp, data in sets.items():
-        for model, keys in data.items():
-            for (key, environment_id), values in keys.items():
+    for timestamp, sets_data in sets.items():
+        for model, sets_keys in sets_data.items():
+            for (key, environment_id), sets_values in sets_keys.items():
```

Comment:
The changes here are basically creating unique variable names for the two loops to help the typing system.